lib/DxcSupport/dxcapi.extval.cpp

```diff
@@ -0,0 +1,61 @@
+#include "dxc/Support/WinIncludes.h"
+#include <dxc/Support/Global.h> // for hresult handling with DXC_FAILED
+#include <filesystem>           // C++17 and later
+// WinIncludes must come before dxcapi.extval.h
+#include "dxc/Support/dxcapi.extval.h"
+
+namespace dxc {
+
+HRESULT DxcDllExtValidationSupport::CreateInstance(REFCLSID clsid, REFIID riid,
+                                                   IUnknown **pResult) {
+  if (DxilExtValSupport.IsEnabled() && clsid == CLSID_DxcValidator)
+    return DxilExtValSupport.CreateInstance(clsid, riid, pResult);
+
+  return DxCompilerSupport.CreateInstance(clsid, riid, pResult);
+}
+
+HRESULT DxcDllExtValidationSupport::CreateInstance2(IMalloc *pMalloc,
```

Comment:
nit: Can we just drop the `2` here and rely on overloading? The param list is different so I think the compiler would be happy